spec/services/cleaners/advocate_claim_cleaner_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'rails_helper'
+require 'services/cleaners/cleaner_shared_examples'
+
+RSpec.shared_examples 'delete fixed fees' do
+  it { expect { call_cleaner }.to change { claim.fixed_fees.size }.to 0 }
+end
+
+RSpec.shared_examples 'does not delete fixed fees' do
+  it { expect { call_cleaner }.not_to(change { claim.fixed_fees.size }) }
+end
+
+RSpec.shared_examples 'does not delete misc fees' do
+  it { expect { call_cleaner }.not_to(change { claim.misc_fees.size }) }
+end
+
+RSpec.shared_examples 'clear basic fees' do
+  it { expect { call_cleaner }.not_to(change { claim.basic_fees.size }) }
+  it { expect { call_cleaner }.to change { claim.basic_fees.flat_map(&:dates_attended).size }.to 0 }
+end
+
+RSpec.shared_examples 'does not clear basic fees' do
+  it { expect { call_cleaner }.not_to(change { claim.basic_fees.size }) }
+  it { expect { call_cleaner }.not_to(change { claim.basic_fees.flat_map(&:dates_attended).size }) }
+end
+
+RSpec.describe Cleaners::AdvocateClaimCleaner do
+  subject(:cleaner) { described_class.new(claim) }
+
+  before do
+    seed_case_types
+    seed_fee_types
+    seed_fee_schemes
+  end
+
+  describe '#call' do
+    subject(:call_cleaner) { cleaner.call }
+
+    let(:claim) do
+      create(
+        :advocate_final_claim, :draft,
+        case_type: build(:case_type, :fixed_fee),
+        fixed_fees: [create(:fixed_fee, :fxase_fee, :with_date_attended, rate: 9.99)]
+      )
+    end
```

Comment:
line break between multiline lets - i would have though rubocop would have picked that up?

Reply:
Added.